tests/wizard/test_forms.py

```diff
@@ -158,6 +158,34 @@ def test_form_condition(self):
         response, instance = testform(request)
         self.assertEqual(instance.get_next_step(), 'step2')
 
+        def previous_step_check(wizard):
+            data = wizard.get_cleaned_data_for_step('start') or {}
+            return not data.get('name', None) == 'SKIP'
+
+        testform = TestWizard.as_view(
+            [('start', Step1), ('step2', Step2), ('step3', Step3)],
+            condition_dict={'step2': previous_step_check}
+        )
+        response, instance = testform(request)
+        self.assertEqual(instance.get_next_step(), 'step2')
+
+        # def subsequent_step_check(wizard):
+        #     data = wizard.get_cleaned_data_for_step('step3') or {}
+        #     return data.get('foo')
+        #
+        # testform = TestWizard.as_view(
+        #     [('start', Step1), ('step2', Step2), ('step3', Step3)],
+        #     condition_dict={'step3': subsequent_step_check}
+        # )
+        # response, instance = testform(request)
```

Comment:
There's an issue with trying to catch infinite recursion in here - because we are already at the maximum recursion depth, so we need to think of a better way of catching them, I think.